src/EdgeConProblem/EdgeConReduction.c

```diff
@@ -8,36 +8,420 @@
+    ctx->G = getGraph(graph);
+    ctx->n = ctx->G.numNodes;
+    ctx->m = ctx->G.numEdges;
+    ctx->C_H = getNumComponents(graph);
+    ctx->N = ctx->C_H - 1;
+    ctx->k = cost;
+    ctx->z3_ctx = z3_ctx;
+
+    return ctx;
+}
+
+
+static Z3_ast build_phi_2(const g_context_s *ctx){
+    int tab1Size = (ctx->m * (ctx->m - 1)) / 2;     // /!\ A REVOIR
+    Z3_ast resTab1[tab1Size];
+    int tabId = 0;
+
+    for (int i = 0; i < ctx->N; i++){
+        for(int e1 = 0; e1 < ctx->m; e1++){             // 1st node of e
+            for(int e2 = e1 + 1; e2 < ctx->m; e2++){    // 2nd node of e
+                for(int j1 = 0; j1 > e1 && j1 < ctx->m; j1++){  // 1st node of j  
+                    for (int j2 = j1 + 1; j2 < ctx->m; j2++){   // 2nd node of j
+                        // j1 > e1 : car on ne test qu'une fois une paire d'arêtes
+                        if(isEdge(ctx->G, e1, e2) && isEdge(ctx->G, j1, j2)){
+                            Z3_ast e_i = X_(e1, e2, i);
+                            Z3_ast j_i = X_(j1, j2, i);
+                            Z3_ast not_e_i = NOT(e_i);
+                            Z3_ast not_j_i = NOT(j_i);
+                            resTab1[tabId++] =  OR(2) 
+                                                    not_e_i,
+                                                    not_j_i
+                                                EOR;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    Z3_ast res1 = Z3_mk_and(ctx->z3_ctx, tabId, (Z3_ast *) resTab1);
+
+    int tab2Size = ((ctx->m) * (((ctx->N) * (ctx->N)) / 2));
+    Z3_ast resTab2[tab2Size];
+    tabId = 0;
+
+    for(int e1 = 0; e1 < ctx->m; e1++){
+        for(int e2 = e1 + 1; e2 < ctx->m; e2++){
+            if(isEdge(ctx->G, e1, e2)){
+                for(int i = 0; i < ctx->N; i++){
+                    for(int j = i + 1; j < ctx->N; j++){
+                        Z3_ast e_i = X_(e1, e2, i);
+                        Z3_ast e_j = X_(e1, e2, j);
+                        Z3_ast not_e_i = NOT(e_i);
+                        Z3_ast not_e_j = NOT(e_j);
+                        resTab2[tabId++] =  OR(2) 
+                                                not_e_i,
+                                                not_e_j
+                                            EOR;
+                    }
+                }
+            } 
+        }
+    }
+
+    Z3_ast res2 = Z3_mk_and(ctx->z3_ctx, tabId, (Z3_ast *) resTab2);
+    Z3_ast res[2] = {res1, res2}; 
+    return Z3_mk_and(ctx->z3_ctx, 2, res);
+}
+
+static Z3_ast build_phi_3(const g_context_s *ctx) { 
+    
+    int tab1Size = (ctx->C_H - 1) * (ctx->C_H - 1);
+    Z3_ast resTab1[tab1Size];
+    int tabId = 0;
+
+    for(int j = 2; j < ctx->C_H; j++){
+        for(int j_prime = j + 1; j_prime < ctx->N; j_prime ++){
+            resTab1[tabId++] = P_(j, j_prime);
+        }
+    }
+
+    Z3_ast res1 = Z3_mk_and(ctx->z3_ctx, tabId, (Z3_ast *) resTab1);
```

Comment:
**issue:** It's not a whole conjunction here. You need to create a disjunction for each `j`.